server/src/immich/app.service.ts

```diff
@@ -28,4 +59,47 @@ export class AppService {
     await this.serverService.handleVersionCheck();
     this.logger.log(`Feature Flags: ${JSON.stringify(await this.serverService.getFeatures(), null, 2)}`);
   }
+
+  ssr(excludePaths: string[]) {
+    const index = readFileSync('/usr/src/app/www/index.html').toString();
+
+    return async (req: Request, res: Response, next: NextFunction) => {
+      if (
+        req.url.startsWith('/api') ||
+        req.method.toLowerCase() !== 'get' ||
+        excludePaths.find((item) => req.url.startsWith(item))
+      ) {
+        return next();
+      }
+
+      const targets = [
+        {
+          regex: /^\/share\/(.+)$/,
+          onMatch: async (matches: RegExpMatchArray) => {
+            const key = matches[1];
```

Comment:
Named capture groups <3

Reply:
Those are too advanced for me. You should fix it :-)

Reply:
Always let the new one do the work... :D